storage/grpc_client.go

```diff
@@ -2152,69 +2202,99 @@ func (d *readResponseDecoder) advanceOffset(n uint64) error {
 // This copies object data from the message into the buffer and returns the number of
 // bytes copied. The data offsets are incremented in the message. The updateCRC
 // function is called on the copied bytes.
-func (d *readResponseDecoder) readAndUpdateCRC(p []byte, updateCRC func([]byte)) int {
-	// For a completely empty message, just return 0
+func (d *readResponseDecoder) readAndUpdateCRC(p []byte, readID int64, updateCRC func([]byte)) (n int, found bool) {
+	// For a completely empty message, just return 0.
 	if len(d.databufs) == 0 {
-		return 0
+		return 0, true
```

Comment:
nit: I find it a bit weird for an empty message to return found as true. I guess this is a case we want to return when being called in Read? Maybe add a comment to clarify?

Reply:
I think actually it would be fine to receive the next message in this case... fixed.